### ISSUE TITLE
resume之后出现Hardfault

### ISSUE THREAD

**Issue by Eplankton** [microcai/mcu_coro_demo#1]
使用demo中的blinky代码进行测试，poll之后的自动resume调用的堆栈好像跳转到了错误的`_M_fr_ptr`，导致core进入Hardfault_handler，使用ARM GCC 13编译。
```
void resume() const { __builtin_coro_resume(_M_fr_ptr); }
```
![图片](https://github.com/user-attachments/assets/ecbea8e5-0d6a-4170-8d3b-f0f1321f9e2e)

**Comment by microcai** [comment]
啊哦。

**Comment by microcai** [comment]
void setup()
{
  led_blinker().detach();
}
改成这个看看。

**Comment by Eplankton** [comment]
> void setup() { led_blinker().detach(); } 改成这个看看。

瞬间搞定，是什么原理呢？感觉这样使用的话会变成一堆：
```
task1().detach();
task2().detach();
task3().detach();
...
```

**Comment by microcai** [comment]
> > void setup() { led_blinker().detach(); } 改成这个看看。
> 
> 瞬间搞定，是什么原理呢？感觉这样使用的话会变成一堆：
> 
> ```
> task1().detach();
> task2().detach();
> task3().detach();
> ...
> ```

哦，那是 ucoro 的内部实现改了。这个demo 是用的以前老版本 awaitable.hpp 的时候写的。最近把  awaitable.hpp 给更新了。没想到这里出问题了。改下就好了。

**Comment by Eplankton** [comment]
> > > void setup() { led_blinker().detach(); } 改成这个看看。
> > 
> > 
> > 瞬间搞定，是什么原理呢？感觉这样使用的话会变成一堆：
> > ```
> > task1().detach();
> > task2().detach();
> > task3().detach();
> > ...
> > ```
> 
> 哦，那是 ucoro 的内部实现改了。这个demo 是用的以前老版本 awaitable.hpp 的时候写的。最近把 awaitable.hpp 给更新了。没想到这里出问题了。改下就好了。

`detach()`的语义应该是没问题的，但是像下面这样写法跑出来的结果总是`led1`先执行完，才能执行`led2`。并行执行的效果似乎消失了：
```C++
mcucoro::awaitable<int> led1(int n)
{
	uint32_t cnt = 0;
	for (int i = 0; i < n; i++) {
		leds[0].toggle();
		cnt += i;
		co_await coro_delay_ms(250);
	}
	LOG("led1 done!");
	co_return cnt;
}

mcucoro::awaitable<int> led2(int n)
{
	uint32_t cnt = 0;
	for (int i = 0; i < n; i++) {
		leds[1].toggle();
		cnt += i;
		co_await coro_delay_ms(250);
	}
        LOG("led2 done!");
	co_return cnt;
}

void setup()
{
	auto test = [] -> mcucoro::awaitable<void> {
		auto s1 = co_await led1(10);
		auto s2 = co_await led2(10);
		LOG("sum=%d", s1 + s2);
	};

	test().detach();

	while (true) {
		mcucoro::executor::system_executor().poll();
	}
}
```

**Comment by microcai** [comment]
你都 co_await led1 了 , 那自然是要等 led1 执行完毕啊！想啥呢。

**Comment by microcai** [comment]
一个 .detach() 就是创建一个协程。你不会以为 co_await 是用来创建协程的吧？
你一共就创建了 test() 这一个协程，还想并发啥呢？

**Comment by Eplankton** [comment]
> 一个 .detach() 就是创建一个协程。你不会以为 co_await 是用来创建协程的吧？ 你一共就创建了 test() 这一个协程，还想并发啥呢？

哈哈，我以为“像写同步阻塞代码一样写异步IO”就是类似的意思。之前一般协程都和`std::future`绑在一起用。
